value.go

```diff
@@ -289,6 +289,20 @@ func tryGetMappedSliceValue[T any](val T, isArray bool, sliceLength int) (mappin
 	if elementLogicType.Ptr == nil {
 		return elementLogicType, mapping.Value{}, unsupportedTypeError(reflect.TypeOf(val).Name())
 	}
+
+	// Validate that all non-null elements have the same type to prevent mixed-type crashes
+	expectedType := mapping.GetTypeId(elementLogicType)
+	for i, lt := range childLogicTypes {
```

Comment:
I wonder if we can move this up into the `range vSlice` loop? Then, we can early-out there already (and don't need to process all parameters first), if we encounter a mismatch.